spec/records_spec.rb

```diff
@@ -0,0 +1,254 @@
+# frozen_string_literal: true
+
+# NOTE: 000004165 is a redirect
+
+VALID_LINE = "#{TEST_HTID}\tdeny\tic\t#{TEST_RECID}\tother fields..."
+INVALID_BYTE_SEQUENCE_LINE = VALID_LINE + "\xFFFF"
+UNPARSEABLE_LINE = ""
+
+RSpec.describe HathifileHistory::Records do
+  let(:records) do
+    recs = described_class.new
```

Comment:
If `logger` were an optional argument to `HathifileHistory::Records`, then this could be 1 line:

`let(:records) { described_class.new(logger: NullLogger.new $stdout) }`

That's at least my personal preference.